### ISSUE TITLE
Cannot reproduce aveP metric

### ISSUE THREAD

**Issue by yanisadel** [f-kretschmer/bertax#22]
Hello,

I used your script to compute predictions of your model on the final_model_dataset/test.fa set.
However, I am not able to reproduce the metrics you obtained.

I am using
```python
sklearn.metrics.average_precision_score(
    y_true_superkingdoms, y_score_superkingdoms, average="macro"
)
```

**However, I am getting 99.77, while 98.62 is the reported value on your paper.**

Would you please have an idea why I am getting this difference ?
Thank you!

**Comment by flomock** [comment]
Its been a while since we implemented this, therefor, take this message with a grain of salt. But it looks for me like you replicated our evaluation score when using Bertax together with MMSeq2.

<img width="920" height="357" alt="Image" src="https://github.com/user-attachments/assets/2839b93a-88ef-4f33-b60c-c9ba189f5405" />

see yellow marked area.

What command did you use to generate the predictions?
